stan/math/prim/mat/fun/mdivide_left_tri.hpp

```diff
@@ -52,6 +58,77 @@ inline Eigen::Matrix<T, R1, C1> mdivide_left_tri(
   return b;
 }
 
+/**
+ * Returns the solution of the system Ax=b when A is triangular
+ * and A and b are matrices of doubles.
+ * @param A Triangular matrix.  Specify upper or lower with TriView
+ * being Eigen::Upper or Eigen::Lower.
+ * @param b Right hand side matrix or vector.
+ * @return x = A^-1 b, solution of the linear system.
+ * @throws std::domain_error if A is not square or the rows of b don't
+ * match the size of A.
+ */
+template <int TriView, int R1, int C1, int R2, int C2>
+inline Eigen::Matrix<double, R1, C2> mdivide_left_tri(
+    const Eigen::Matrix<double, R1, C1> &A,
+    const Eigen::Matrix<double, R2, C2> &b) {
+  check_square("mdivide_left_tri", "A", A);
+  check_multiplicable("mdivide_left_tri", "A", A, "b", b);
+#ifdef STAN_OPENCL
+  if (A.rows()
+      >= opencl_context.tuning_opts().lower_tri_inverse_size_worth_transfer) {
+    matrix_cl A_cl(A);
+    matrix_cl b_cl(b);
+    matrix_cl A_inv_cl(A.rows(), A.cols());
+    if (TriView == Eigen::Lower) {
+      A_inv_cl = lower_triangular_inverse(A_cl);
+    } else {
+      A_inv_cl = transpose(lower_triangular_inverse(transpose(A_cl)));
+    }
```

Comment:
How do you feel about making a templated `triangular_inverse`? Doesn't have to be in this PR but could be nice to have something like `triangular_inverse<TriView>(A_cl);`.

Where Lower would do what we have for `lower_triangular_inverse` and Upper could either do the transpose first or call a kernel with indices for the upper

Reply:
Yeah, I will add the templated function here. The PR isnt really huge ATM so I will add the templated version with transposing for now and make an issue to make a upper_triangular_inverse kernel.